### ISSUE TITLE
Amazon EKS Cluster OIDC Issuer URL

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#9995]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Amazon EKS now allows you to assign IAM permissions to Kubernetes service accounts.
This is achieved via an OIDC Issuer URL exposed on an EKS Cluster.

### New or Affected Resource(s)

* [`aws_eks_cluster`](https://www.terraform.io/docs/providers/aws/r/eks_cluster.html)
### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2019/09/amazon-eks-adds-support-to-assign-iam-permissions-to-kubernetes-service-accounts/).
[API reference](https://docs.aws.amazon.com/eks/latest/APIReference/API_Identity.html).
[Blog post](https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/).
[AWS containers roadmap issue](https://github.com/aws/containers-roadmap/issues/23).

Requires [AWS SDK v1.23.15](https://github.com/aws/aws-sdk-go/releases/tag/v1.23.15):
* https://github.com/terraform-providers/terraform-provider-aws/pull/9970

**Comment by bflad** [comment]
I will have the new attributes and an example submitted shortly. 👍

**Comment by bflad** [comment]
Pull request submitted: https://github.com/terraform-providers/terraform-provider-aws/pull/10006

**Comment by marcincuber** [comment]
> Pull request submitted: #10006

Can we please get that reviewed and merge asap ;)? I can finally remove kube2iam

**Comment by bflad** [comment]
The functionality to retrieve the OIDC issuer URL from the `aws_eks_cluster` resource and data source has been merged and will release with version 2.28.0 of the Terraform AWS Provider, on Thursday. 👍
